src/lib.rs

```diff
@@ -0,0 +1,389 @@
+//! `fm` is a simple non-backtracking fuzzy text matcher useful for matching multi-line patterns
+//! and text. At its most basic the wildcard operator (`...` by default) can be used in the
+//! following ways:
+//!
+//!   * If a line consists solely of `...` it means "match zero or more lines of text".
+//!   * If a line starts with `...` it means "match the remainder of the line only".
```

Comment:
I find the wording of this confusing:

> If a line starts with `...` it means "match the remainder of the line only

We always match the remainder of the line, even in the absence of `...`.

I think I know what you mean, but I wonder if a more direct wording would be clearer.

Why not collapse the last three cases into something like:

> If a line does not consist solely of `...` then instances of `...` perform greedy(?) elision within the line.

Or:

> By default, matches are anchored to the start and end of the line. This can be relaxed by starting or ending (respectively) a line with `...`.

Reply:
The difficulty is that `...` is only a wildcard at the start or end of a line. I'm open to phrasing suggestions, but fundamentally, it is a bit awkward (because we're not backtracking).

Reply:
How about this idiom:
> If a line starts with `...` it the match is not anchored to the start of the line.

Reply:
Yes, I think that might work. What do you think of https://github.com/softdevteam/fm/pull/1/commits/1d296a50db23cba6802a23c269759849fcc1bc67?

Reply:
I think that's good!

Is it worth saying somewhere that by default (unless you use wildcards) the matches are anchored? It's *implied* already, but is it better to be explicit?